o11y/metrics/metrics.go

```diff
@@ -49,6 +49,7 @@ type Sink interface {
 	MetricTooLate(tags []string) error
 	MetricTargetMissing(duration time.Duration, tags []string) error
 	MetricMissingTargetFound(tags []string) error
+	MetricMissingTargetDeleted(tags []string) error
```

Comment:
we a growing list of  metrics now... Should we rename them to be more explicit on its usage?
For example in this file we have `MetricCleaned` but it could be confusing. 

And I'd also like to move the comment declaration on the struct itself as well, so that in one look we understand each metric. 

It's probably unrealistic to have both the code and external documentation on each metric we emit so if we have a very clear code file it's fine with me!

Reply:
I do think having the comments here would be nice. Would we want to duplicate them, or just move them out of the implementation code? renaming everything isn't bad, but that's another PR

Reply:
I think duplicating is better? But then we'll need to remember to keep both files updated on comments.